packages/siop-oid4vp/lib/authorization-request/Payload.ts

```diff
@@ -18,20 +18,28 @@ import { createRequestRegistration } from './RequestRegistration'
 import { ClaimPayloadOptsVID1, CreateAuthorizationRequestOpts, PropertyTarget } from './types'
 
 export const createPresentationDefinitionClaimsProperties = (opts: ClaimPayloadOptsVID1): ClaimPayloadVID1 => {
-  if (!opts || !opts.vp_token || (!opts.vp_token.presentation_definition && !opts.vp_token.presentation_definition_uri)) {
+  if (
+    !opts ||
+    !opts.vp_token ||
+    (!opts.vp_token.presentation_definition && !opts.vp_token.presentation_definition_uri && !opts.vp_token.dcql_query)
+  ) {
     return undefined
   }
-  const discoveryResult = PEX.definitionVersionDiscovery(opts.vp_token.presentation_definition)
-  if (discoveryResult.error) {
-    throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID)
+
+  if (opts.vp_token.presentation_definition || opts.vp_token.presentation_definition_uri) {
+    const discoveryResult = PEX.definitionVersionDiscovery(opts.vp_token.presentation_definition)
```

Comment:
Cannot be correct in case a definitition_uri is passed in. These are xor, and you are only looking for the actual definition object here

Reply:
This comes from the original PR: https://github.com/Sphereon-Opensource/OID4VC/blob/ad19797805d68855d902afcf586d521927654adc/packages/siop-oid4vp/lib/authorization-request/Payload.ts#L29-L34

Reply:
Yes?! You made it your code by copy pasting it. It is introduced in this PR, so it needs to be fixed

Reply:
I think you got me wrong. From the project management perspective it makes more sense to have the person who wrote the code to fix its issues because it's more efficient. (At the point I've wrote the comment above, there was an open PR and it was even reviewed). I should have only started to work on this code after the original PR was merged to avoid dependency on tickets, to reduce complexity and conflicts.

Reply:
It's addressed

Reply:
No I didn't get it wrong. This is new code being I produced in this PR. If there then is an issue, you simply cannot say.. Not my problem

Reply:
fixed

Reply:
I'll approve while waiting for @nklomp  input here

Reply:
> content of presentation_definition_uri and do a discovery on that? (It did not do that before)

Yes see above. Removing the URI version is play wrong. We already have a function to do this. So if it is a URI just fetch is and then you will end up with a the non URI version